pkg/infrastructure/powervs/clusterapi/powervs.go

```diff
@@ -163,6 +163,34 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 	if err != nil {
 		return fmt.Errorf("failed to add ping security group rule: %w", err)
 	}
+
+	if in.InstallConfig.Config.Publish == types.InternalPublishingStrategy &&
+		(len(in.InstallConfig.Config.ImageDigestSources) > 0 || len(in.InstallConfig.Config.DeprecatedImageContentSources) > 0) {
+		var client *powervsconfig.Client
+		client, err = powervsconfig.NewClient()
+		if err != nil {
+			return fmt.Errorf("failed to obtain PowerVS client: %w", err)
+		}
+
+		name := fmt.Sprintf("%s-vpe-iam", in.InfraID)
+		vpcID := *powerVSCluster.Status.VPC.ID
+		logrus.Debugf("InfraReady: Ensuring VPE gateway for IAM in VPC %v", vpcID)
+		rgID := *powerVSCluster.Status.ResourceGroup.ID
+		subnetID := *powerVSCluster.Status.VPCSubnet[*powerVSCluster.Spec.VPCSubnets[1].Name].ID
+		_, err = client.CreateVirtualPrivateEndpointGateway(ctx, name, vpcID, subnetID, rgID, "crn:v1:bluemix:public:iam-svcs:global:::endpoint:private.iam.cloud.ibm.com")
```

Comment:
This isn't needed for the private case, it's needed for the disconnected case. So there should be an inner check here that ImageContentDigests (or sources) is set.

Reply:
Reading closer this seems to be just a test, so you probably are going to make this cleaner once you start considering other VPEs that are required for disconnected :)

Reply:
Line 168 in the proposed change is what I _think_ signifies the Disconnected install, on top of line 167 checking it's an `Internal` (=private) install. So, this is real code that'd create IAM VPE for Disconnected install, though it'd also execute, if you specify `imageContentSource` pointing to public `quay.io` repos as your "mirror," and it'd perhaps fail. So, that's the early discussion point I'm looking to have. 🙂

Reply:
ah sorry i see it now